client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * Test to see if the fragment is reached when tapped
+     */
+    @Test
+    public void MyProfileFragmentTapReachedTest() {
+        // Navigate to the user profile fragment.
+        solo.clickOnView(solo.getView(R.id.bottom_navigation_item_profile));
+        assertTrue(solo.waitForText("Profile", 1, 2000));
+    }
+
+    /**
+     * Test to see if the bio text is editable
+     */
+    @Test
+    public void editableBioTextTest() {
+        solo.scrollToSide(Solo.RIGHT);
+        assertTrue(solo.waitForText("Profile", 1, 2000));
+        solo.clickOnButton("Edit");
+        String testBioString = "This is a bio text test";
+        solo.clearEditText((EditText) solo.getView(R.id.bio_text));
+        assertFalse(solo.waitForText(testBioString, 1, 2000));
+        solo.enterText((EditText) solo.getView(R.id.bio_text), testBioString);
+        assertTrue(solo.waitForText(testBioString, 1, 2000));
+    }
+
+    /**
+     * Test to see if the phone text is editable
+     */
+    @Test
+    public void editablePhoneTextTest() {
+        solo.scrollToSide(Solo.RIGHT);
+        assertTrue(solo.waitForText("Profile", 1, 2000));
+        solo.clickOnButton("Edit");
+        String testPhoneString = "9999999999";
+        solo.clearEditText((EditText) solo.getView(R.id.phone_text));
+        assertFalse(solo.waitForText(testPhoneString, 1, 2000));
+        solo.enterText((EditText) solo.getView(R.id.phone_text), testPhoneString);
+        assertTrue(solo.waitForText(testPhoneString, 1, 2000));
+    }
+
+    /**
+     * Test to see if the address text is editable
+     */
+    @Test
+    public void editableAddrTextTest() {
+        solo.scrollToSide(Solo.RIGHT);
+        assertTrue(solo.waitForText("Profile", 1, 2000));
+        solo.clickOnButton("Edit");
+        String testAddrString = "99 Wayne Gretzky Ave";
+        solo.clearEditText((EditText) solo.getView(R.id.phone_text));
+        assertFalse(solo.waitForText(testAddrString, 1, 2000));
+        solo.enterText((EditText) solo.getView(R.id.phone_text), testAddrString);
+        assertTrue(solo.waitForText(testAddrString, 1, 2000));
+    }
+
+    /**
+     * Test to see if the fragment lets the user cancel the edit
+     */
+    @Test
+    public void editTextCancelTest() {
+        solo.scrollToSide(Solo.RIGHT);
+        assertTrue(solo.waitForText("Profile", 1, 2000));
+        solo.clickOnButton("Edit");
+        String testString = "This is a bio text test";
+        solo.clearEditText((EditText) solo.getView(R.id.bio_text));
+        assertFalse(solo.waitForText(testString, 1, 2000));
+        solo.enterText((EditText) solo.getView(R.id.bio_text), testString);
+        assertTrue(solo.waitForText(testString, 1, 2000));
+        solo.clickOnButton("Cancel");
+        assertFalse(solo.waitForText(testString, 1, 2000));
+    }
+
+    /**
+     * Test to see if the fragment lets the user save the edit
+     */
+    @Test
+    public void editTextSaveTest() {

```

Comment:
are you reverting this after? once real saves come in you should revert it to bio